src/d3-ext/profile.js

```diff
@@ -0,0 +1,320 @@
+   */
+  var bisectDistance = d3.bisector(function(d) { return d['dist']; }).left;
+
+  /**
+   * Distance units. Either 'm' or 'km'.
+   */
+  var units;
+
+  /**
+   * Factor to determine whether to use 'm' or 'km'.
+   */
+  var xFactor;
+
+  /**
+   * @type {function(Object)}
+   * Hover callback function.
+   */
+  var hover = function() {};
+
+  /**
+   * @type {function()}
+   * Out callback function.
+   */
+  var out;
+
+  /**
+   * The color to be used for filling the area.
+   * Can be overriden using the '.area' CSS selector.
+   * @type {string}
+   */
+  var FILL_COLOR = 'rgba(222, 222, 222, 0.5)';
+
+  /**
+   * The color to be used the line stroke.
+   * Can be overriden using the '.line' CSS selector.
+   * @type {string}
+   */
+  var STROKE_COLOR = '#F00';
+
+  /**
+   * @type {Object}
+   */
+  var svg;
+
+  /**
+   * D3 x scale.
+   */
+  var x;
+
+  /**
+   * D3 y scale.
+   */
+  var y;
+
+  var profile = function(selection) {
+    selection.each(function(data) {
+
+      var width = this.getBoundingClientRect().width -
+          margin.right - margin.left;
+      x = d3.scale.linear().range([0, width]);
+
+      var height = this.getBoundingClientRect().height -
+          margin.top - margin.bottom;
+      y = d3.scale.linear().range([height, 0]);
+
+      var xAxis = d3.svg.axis().scale(x).orient('bottom');
+      var yAxis = d3.svg.axis()
+          .scale(y)
+          .orient('left');
+
+      var area = d3.svg.area()
+          .x(function(d) { return x(d['dist']); })
+          .y0(height)
+          .y1(function(d) { return y(d['values']['mnt']); });
+      var line = d3.svg.line()
+          .x(function(d) { return x(d['dist']); })
+          .y(function(d) { return y(d['values']['mnt']); });
+
+      // Select the svg element, if it exists.
+      svg = d3.select(this).selectAll('svg').data([data]);
+
+      // Otherwise, create the skeletal chart.
+      var gEnter = svg.enter().append('svg').append('g');
+      gEnter.style('font', '11px Arial');
+      gEnter.append('path').attr('class', 'area')
+          .style('fill', FILL_COLOR);
+      gEnter.append('path').attr('class', 'line')
+          .style('stroke', STROKE_COLOR)
+          .style('fill', 'none');
+
+      gEnter.insert('g', ':first-child')
+          .attr('class', 'grid-y');
+
+      gEnter.append('g')
+          .attr('class', 'x axis')
+          .attr('transform', 'translate(0,' + height + ')');
+
+      gEnter.append('text')
+          .attr('class', 'x label')
+          .attr('text-anchor', 'end')
+          .attr('x', width - 4)
+          .attr('y', height - 4);
+
+      gEnter.append('g')
+          .attr('class', 'y axis');
+
+      gEnter.append('text')
+          .attr('class', 'y label')
+          .attr('text-anchor', 'end')
+          .attr('y', 6)
+          .attr('dy', '.75em')
+          .attr('transform', 'rotate(-90)')
+          .style('fill', 'grey')
+          .text('elevation (m)');
```

Comment:
Do you mean the texts are currently set once at creation time?
When is a text update needed?